whylog/tests/test_regex_assistant/__init__.py

```diff
@@ -1,57 +1,115 @@
+import re
 from unittest import TestCase
 
-from nose.plugins.skip import SkipTest
-
-from whylog.assistant import RegexAssistant
-from whylog.assistant.regex_assistant_exceptions import NoDateGroupException
+from whylog.assistant.regex_assistant import RegexAssistant
+from whylog.assistant.regex_assistant.regex import (
+    NotMatchingRegexError, create_date_regex, create_obvious_regex, verify_regex
+)
+from whylog.assistant.span import sort_by_start
+from whylog.assistant.spans_finding import (
+    _find_date_spans_by_force, _find_spans_by_regex, find_date_spans
+)
 from whylog.front import FrontInput
-from whylog.teacher import Interval
 
 
 class TestBasic(TestCase):
-    def test_one(self):
-        content1 = "2015-12-03 12:08:09 Connection error occurred on beta36"
-        content2 = "2015-12-03 12:10:10 Data migration from beta36 to beta21 failed"
-        content3 = "2015-12-03 12:11:00 Data is missing on beta21"
+    def test_verify_regex(self):
+        line = r"2015-12-03 12:11:00 Data is missing on comp21"
+        regex = r"^(\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d) Data is missing on (.*)$"
+        matched, groups, errors = verify_regex(regex, line)
+        assert (matched, len(groups), len(errors)) == (True, 2, 0)
+        assert groups[0] == r"2015-12-03 12:11:00"
+        assert groups[1] == r"comp21"
+
+        line2 = r"2015-12-03 12:11:00 Data is missing"
+        matched, groups, errors = verify_regex(regex, line2)
+        assert (matched, len(groups), len(errors)) == (False, 0, 1)
+        assert isinstance(errors[0], NotMatchingRegexError)
+
+    def test_create_obvious_regex(self):
+        line = r".^$*x+x{5}?\*[x]x|y(x)(?iLmsux)(?:x)(?P<name>x)(?#x)(?<!x)\4\b\A"
+        obvious_regex = create_obvious_regex(line)
+        assert obvious_regex == \
+               r"\.\^\$\*x\+x\{5\}\?\\\*\[x\]x\|y\(x\)\(\?iLmsux\)\(\?:x\)\(\?P<name>x\)\(\?#x\)\(\?<!x\)\\4\\b\\A"
+        matched, groups, errors = verify_regex(obvious_regex, line)
+        assert (matched, len(groups), len(errors)) == (True, 0, 0)
 
-        cause1 = FrontInput(903, content1, None)
-        cause2 = FrontInput(53, content2, None)
-        effect = FrontInput(1087, content3, None)
+    def test_create_date_regex(self):
+        date = '10/Oct/1999:21:15:05'
+        regex = create_date_regex(date)
+        matched, groups, errors = verify_regex(regex, date)
+        assert (matched, len(groups), len(errors)) == (True, 0, 0)
+        bad_dates = [
+            date + " ", date + ":", "1" + date, '10/10/1999:21:15:05', '10/Oct/199:21:15:05',
```

Comment:
These are good date formates. It is not a test for being good date or not (It is test for matching the regex created from string: '10/Oct/1999:21:15:05'). Yes, I assume that date format in every instances of some log are the same. Otherwise date formats would be more like (.*), because we should be prepared for every possible date format then.